extensions/package.json

```diff
@@ -3,7 +3,7 @@
   "version": "0.0.1",
   "description": "Dependencies shared by all extensions",
   "dependencies": {
-    "typescript": "3.7.3"
+    "typescript": "3.8.0-dev.20200104"
```

Comment:
I'm unsure what the normal procedure is for referencing newer builds of TypeScript, but the insiders build referenced here is needed for the protocol API changes in the TS Server.

Reply:
We only ship stable TS versions in stable VS Code. That means that we can't update this until TS 3.8 is released in mid February.

To handle cases like this, we instead add forward declarations of the required TSServer types. That way users can start using a newer TS version to try out this feature if they want. But we keep the actually shipped TS version on the current stable build

Does that make sense?